AI-and-Analytics/End-to-end-Workloads/LidarObjectDetection-PointPillars/src/main.cpp

```diff
@@ -0,0 +1,133 @@
+//==============================================================
+// Copyright © 2020-2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <boost/range/iterator_range.hpp>
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include "pointpillars/pointpillars_config.hpp"
+#include "pointpillars/pointpillars_util.hpp"
+#include "pointpillars/pointpillars.hpp"
+#include "devicemanager/devicemanager.hpp"
+
+std::size_t ReadPointCloud(std::string const &file_name, std::vector<float> &points) {
+  if (!boost::filesystem::exists(file_name) || file_name.empty()) {
+    return 0;
+  }
+
+  std::size_t number_of_points = 0;
+
+  std::ifstream in(file_name);
+  std::string line;
+  bool parse_data = false;
+  while (std::getline(in, line) && points.size() <= 4 * number_of_points) {
+    if (parse_data) {
+      std::istringstream iss(line);
+      float x, y, z, intensity;
+      double timestamp;
+      if (!(iss >> x >> y >> z >> intensity >> timestamp)) {
+        return 0;
+      }
+      points.push_back(x);
+      points.push_back(y);
+      points.push_back(z);
+      points.push_back(intensity);
+    } else if (line.find("POINTS") != std::string::npos) {
+      number_of_points = atoll(line.substr(7).c_str());
+    } else if (line.find("DATA") != std::string::npos) {
+      parse_data = true;
+    }
+  }
+
+  return number_of_points;
+}
+
+int main(int argc, char *argv[]) {
+  boost::program_options::options_description desc("Allowed options");
```

Comment:
The boost library is a new dependence I have not seen in other sample code I've reviewed.  Please check to see if we have it approved in the IP plan.

Reply:
The use of boost was approved and is also mentioned in the 3rd-party-licenses

Reply:
Correct